vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -766,13 +952,56 @@ def add_remote_agent(self,
                 self.tp_rank)
 
             # Register with NIXL.
-            descs = self.nixl_wrapper.get_xfer_descs(blocks_data, "VRAM")
+            descs = self.nixl_wrapper.get_xfer_descs(blocks_data,
+                                                     self.nixl_memory_type)
             self.dst_xfer_side_handles[
                 engine_id] = self.nixl_wrapper.prep_xfer_dlist(
                     remote_agent_name, descs)
 
         return remote_agent_name
 
+    def sync_recved_kv_to_device(self,
+                                 req_id: str,
+                                 meta: Optional[ReqMeta] = None):
+        """copy recved kv from host buffer to device."""
+        if not self.use_host_buffer:
+            return
+        assert self.copy_blocks is not None
+
+        if meta is None:
+            meta = self._recving_metadata.get(req_id)
+        if meta and req_id not in self._recving_transfers:
+            # local decode only
+            if not meta.do_remote_prefill:
+                return
+            local_block_ids = meta.local_block_ids
+            self.copy_blocks(self.host_xfer_buffers, self.device_kv_caches,
+                             local_block_ids, local_block_ids, "h2d")
+            logger.debug(
+                "synced recved kv of request[%s] to device kv buffer,"
+                "local_block_ids: %s. ", req_id,
+                ",".join(map(str, meta.local_block_ids)))
```

Comment:
Best to guard these with `if logger.isEnabledFor(logging.DEBUG):` to avoid nontrivial parameter computation